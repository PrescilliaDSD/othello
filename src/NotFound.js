import React from 'react'
import { Link } from 'react-router-dom'

const NotFound = () => {
  return (
    <main className="home">
      <p className="what-to-do">Oups ! Je crois que tu es perdu.</p>
      <Link to="/">Retourner sur la page d&apos;accueil</Link>
    </main>

  )
}

export default NotFound