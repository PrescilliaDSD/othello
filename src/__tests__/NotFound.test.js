import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'

import NotFound from '../components/NotFound'

describe('NotFound component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a main tag', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree.type).toBe('main')
  })

  it('should render a p tag in the main tag', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree.children[0].type).toBe('p')
  })
  it('should render content \'Oups ! Je crois que tu es perdu.\' in the p tag', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree.children[0].children[0]).toBe('Oups ! Je crois que tu es perdu.')
  })
  it('should render a link', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree.children[1].type).toBe('a')
  })

  it('should render a link to the home page', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree.children[1].props.href).toBe('/')
  })

  it('should render a link with content \'Retourner sur la page d\'accueil\' ', () => {
    const tree = renderer
      .create(<Router><NotFound /></Router>)
      .toJSON()
    expect(tree.children[1].children[0]).toBe('Retourner sur la page d\'accueil')
  })
})
