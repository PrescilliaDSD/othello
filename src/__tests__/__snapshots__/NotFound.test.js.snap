// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render correctly 1`] = `
<main
  className="home"
>
  <p
    className="what-to-do"
  >
    Oups ! Je crois que tu es perdu.
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    Retourner sur la page d'accueil
  </a>
</main>
`;
